sdk/eventhub/Azure.Messaging.EventHubs/src/Consumer/CheckpointPosition.cs

```diff
@@ -0,0 +1,137 @@
+        {
+            Offset = offset;
+            SequenceNumber = sequenceNumber;
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="CheckpointPosition"/> from an <see cref="EventData"/> instance.
+        /// </summary>
+        ///
+        /// <param name="eventData">The <see cref="EventData"/> to use to determine the starting point of a checkpoint, indicating that an event processor should begin reading from the next event in the stream.</param>
+        ///
+        public static CheckpointPosition FromEvent(EventData eventData)
+        {
+            return new CheckpointPosition(eventData.SequenceNumber, eventData.Offset);
+        }
+
+        /// <summary>
+        ///   Determines whether the specified <see cref="CheckpointPosition" /> is equal to this instance.
+        /// </summary>
+        ///
+        /// <param name="other">The <see cref="CheckpointPosition" /> to compare with this instance.</param>
+        ///
+        /// <returns><c>true</c> if the specified <see cref="CheckpointPosition" /> is equal to this instance; otherwise, <c>false</c>.</returns>
+        ///
+        public bool Equals(CheckpointPosition other)
+        {
+            return (Offset == other.Offset)
+                && (SequenceNumber == other.SequenceNumber);
+        }
+
+        /// <summary>
+        ///   Determines whether the specified <see cref="System.Object" /> is equal to this instance.
+        /// </summary>
+        ///
+        /// <param name="obj">The <see cref="System.Object" /> to compare with this instance.</param>
+        ///
+        /// <returns><c>true</c> if the specified <see cref="System.Object" /> is equal to this instance; otherwise, <c>false</c>.</returns>
+        ///
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override bool Equals(object obj) =>
+            obj switch
+            {
+                CheckpointPosition other => Equals(other),
+                _ => false
+            };
+
+        /// <summary>
+        ///   Returns a hash code for this instance.
+        /// </summary>
+        ///
+        /// <returns>A hash code for this instance, suitable for use in hashing algorithms and data structures like a hash table.</returns>
+        ///
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override int GetHashCode()
+        {
+            var hashCode = new HashCodeBuilder();
+            hashCode.Add(Offset);
+            hashCode.Add(SequenceNumber);
+
+            return hashCode.ToHashCode();
+        }
+
+        /// <summary>
+        ///   Converts the instance to string representation.
+        /// </summary>
+        ///
+        /// <returns>A <see cref="System.String" /> that represents the position in the event stream.</returns>
+        ///
+        public override string ToString()
+        {
+            var stringRepresentation = $"Sequence Number: [{SequenceNumber}] | Offset: [{Offset}]";
```

Comment:
nit: No need for the intermediate variable, we can just return the expression directly.

Reply:
It's easier to conditionally add the replication segment this way but I can remove it until we add the replication segment back in.